src/CIFAR100/interface.jl

```diff
@@ -0,0 +1,394 @@
+        end
+    end
+    # optionally transform the image array before returning
+    bytes_to_type(T, images::Array{UInt8,4}), labels_c::Vector{Int}, labels_f::Vector{Int}
+end
+
+"""
+    testdata([T = N0f8], [indices]; [dir]) -> X, Yc, Yf
+
+Returns the CIFAR-100 **testset** corresponding to the given
+`indices` as a three-element tuple. If `indices` is omitted the
+full testset is returned. The first element of three return
+values (`X`) will be the images as a multi-dimensional array, the
+second element (`Yc`) the corresponding coarse labels as
+integers, and the third element (`Yf`) the fine labels
+respectively.
+
+The image(s) is/are returned in the native horizontal-major
+memory layout as a single numeric array of eltype `T`. If `T <:
+Integer`, then all values will be within `0` and `255`, otherwise
+the values are scaled to be between `0` and `1`. The integer
+values of the labels correspond 1-to-1 the digit that they
+represent.
+
+```julia
+X, Yc, Yf = CIFAR100.testdata() # full datatset
+X, Yc, Yf = CIFAR100.testdata(dir="./CIFAR100") # custom folder
+x, yc, yf = CIFAR100.testdata(2) # only second observation
+```
+
+$(download_docstring("CIFAR100", DEPNAME))
+
+Take a look at [`CIFAR100.testtensor`](@ref) and
+[`CIFAR100.testlabels`](@ref) for more information.
+"""
+function testdata(args...; dir = nothing)
+    testdata(N0f8, args...; dir = dir)
+end
+
+function testdata(::Type{T}; dir = nothing) where T
+    file_path = datafile(DEPNAME, TESTSET_FILENAME, dir)
+    # simply read the complete content of the testset file
+    images, labels_c, labels_f = Reader.readdata(file_path, TESTSET_SIZE)
+    # optionally transform the image array before returning
+    bytes_to_type(T, images), labels_c, labels_f
+end
+
+function testdata(::Type{T}, index::Integer; dir = nothing) where T
+    @assert 1 <= index <= TESTSET_SIZE "parameter \"index\" ($index) not in 1:$(TESTSET_SIZE)"
+    file_path = datafile(DEPNAME, TESTSET_FILENAME, dir)
+    # read the single image+labelc+labelf corresponding to "index"
+    image, label_c, label_f = Reader.readdata(file_path, TESTSET_SIZE, index)
+    # optionally transform the image array before returning
+    bytes_to_type(T, image), label_c, label_f
+end
+
+function testdata(::Type{T}, indices::AbstractVector; dir = nothing) where T
+    mi, ma = extrema(indices)
+    @assert mi >= 1 && ma <= TESTSET_SIZE "not all elements in parameter \"indices\" are in 1:$(TESTSET_SIZE)"
+    # preallocate a buffer we will reuse for reading individual
+    # images. "buffer" is written to length(indices) times
+    buffer = Array{UInt8,3}(Reader.NROW, Reader.NCOL, Reader.NCHAN)
+    # we know the types and dimensions of the return values,
+    # so we can preallocate them
+    images = Array{UInt8,4}(Reader.NROW, Reader.NCOL, Reader.NCHAN, length(indices))
+    labels_c = Array{Int,1}(length(indices))
+    labels_f = Array{Int,1}(length(indices))
+    file_path = datafile(DEPNAME, TESTSET_FILENAME, dir)
```

Comment:
I wonder if you want to add just the this module(/each module):
`datafile(TESTSET_FILENAME, dir) = datafile(DEPNAME, TESTSET_FILENAME, dir)`
kinda link you did for `download`.